compiler/luci/pass/src/FoldShapePass.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldShapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+namespace
+{
+
+template <loco::DataType OutType> luci::CircleConst *folding(luci::CircleShape *shape)
+{
+  auto input_node = loco::must_cast<luci::CircleNode *>(shape->input());
+  auto name = input_node->name();
+  auto shape_status = input_node->shape_status();
+  auto rank = input_node->rank();
+  assert(name.length() > 0);
+  if (shape_status != luci::ShapeStatus::VALID)
+    return nullptr;
+  if (rank <= 0)
+    return nullptr;
+  for (uint32_t i = 0; i < rank; i++)
+  {
+    auto dim = input_node->dim(i);
+    if (!dim.known()) // If dim is unknown, return nullptr
+      return nullptr;
+  }
+
+  auto folded_shape = input_node->graph()->nodes()->create<luci::CircleConst>();
+  folded_shape->name(name + "_ConstShape");
+  folded_shape->dtype(OutType);
+  folded_shape->rank(1);
+  folded_shape->dim(0).set(rank);
+  luci::add_origin(folded_shape, luci::get_origin(shape));
+
+  folded_shape->size<OutType>(rank);
+  for (uint32_t i = 0; i < rank; i++)
+    folded_shape->at<OutType>(i) = input_node->dim(i).value();
+
+  return folded_shape;
+}
+
+// Fold Shape to const if the input shape is fixed
```

Comment:
q) what is the meaning of `fixed` ?

Reply:
It seems more proper to say `valid`
Sorry for confusion 😅

Reply:
`valid` also looks vague . 
or can you explain the meaning of `valid` here?

Reply:
"whose shape is known, not dynamic" and "VALID" have different points of view.

please search the internet for shape and stuff how others represent and how our codes are implemented.

Reply:
Sorry for the vague explanation

I'll find more explicit expression refering your advice :)

Reply:
I think that **static** is proper to explain what I want to say

below is the meaning of each word what I understand

**static shape** -

Reply:
I think that **static** is proper word to explain what I want to say

---
below is the meaning of the words what I understand

**static shape** - the shape that is determinized at graph definition time (fully-known shape)
It can folded as a constant before run the graph
(e.g. [1, 8, 8, 64])

**dynamic shape** - the shape that is determinized at graph execution time (partially-known shape)
It cannot folded as a constant because we don't know the output of the **shape** node before run the graph
(e.g. [None, 8, 8, 64])

**valid shape** - the shape is known (fully-known or partially known)
This term includes **static shape** and **dynamic shape**
If we don't know any of the dimension value, it is unknown shape (e.g. [None, None, None, None])

Reply:
I think that **static** is proper word to explain what I want to say

---
below is the meaning of the words what I understand

**static shape ** - the shape that is determinized at graph definition time (fully-known shape)
It can folded as a constant before run the graph
(e.g. [1, 8, 8, 64])

**dynamic shape** - the shape that is determinized at graph execution time (partially-known shape / unknown shape / unknown shape and rank)
It cannot folded as a constant because we don't know the output of the **shape** node before run the graph
(e.g. [None, 8, 8, 64])

**valid shape** - the shape is known (fully-known or partially known)